src/Bicep.Core/Intermediate/ExpressionBuilder.cs

```diff
@@ -741,10 +740,81 @@ void AddLoop(LoopExpressionContext newLoop)
             dependencies);
     }
 
-    private ObjectExpression ConvertObject(ObjectSyntax @object)
-        => new(
-            @object,
-            @object.Properties.Select(ConvertObjectProperty).ToImmutableArray());
+    private Expression ConvertArray(ArraySyntax array)
+    {
+        var hasSpread = false;
+        var chunks = new List<Expression>();
+        var currentChunk = new List<Expression>();
+        void completePreviousChunk()
+        {
+            if (currentChunk.Count > 0)
+            {
+                chunks.Add(new ArrayExpression(array, currentChunk.ToImmutableArray()));
+                currentChunk.Clear();
+            }
+        }
+
+        foreach (var child in array.Children)
+        {
+            if (child is SpreadExpressionSyntax spread)
+            {
+                completePreviousChunk();
+                chunks.Add(ConvertWithoutLowering(spread.Expression));
```

Comment:
Would it be worth doing optimizations to drop `...[]` and `...{}`?

Reply:
I think the optimization would only allow us to solve the simplest case (user literally writes `...[]`), because we don't do any constant folding. I can't think of a reason why this pattern would be common, so it feels perhaps not worth the effort to optimize.

Reply:
That makes sense. I agree that this pattern is rare. The only scenario where ...[] and ...{} might be useful could be within a ternary operator. However, in such cases, optimizations are generally not feasible.